ADR/0004-component-image-location.md

```diff
@@ -0,0 +1,107 @@
+# 4. Out-of-the-box image repository for AppStudio users 
+
+Date: Oct 29, 2022 
+
+## Status
+
+Accepted
+
+## Approvers
+
+* Alexey Kazakov
+* Gorkem Ercan
+
+## Reviewers
+
+* Alexey Kazakov
+* Gorkem Ercan
+* Parag Dave
+
+
+## Context
+
+**Problem**
+AppStudio does not have a internal registry where images could be pushed to as an intermediate step before being deployed as a container.
+As an application centric experience, AppStudio should not make it mandatory for its users to specify where the image ( built from source code ) needs
+to be pushed to.
+
+
+**Goals**: 
+* Provide an out-of-the-box location for users’ images to be pushed after being built from source.
+
+
+**Non-Goals**: 
+* Define the user experience for users bringing existing images from other image registry services.
+* Provide users an option to choose what the out-of-the-box location for images would be. 
+* Define the user experience for users who are willing to configure additional credentials for pushing to an image registry of their choice.
+
+**Design Goals**
+* Use Quay.io as the out-of-the-box image registry in order to avoid getting into the business of maintaining an internal registry.
+* Align user permissions in an AppStudio workspace with those in Quay.io
+* Maintain the right levels of isolation between images being expected of a multi-tenant system.
+* Security:
+  * Leverage short-lived Quay.io API tokens for repo/org management on Quay.io
+  * Leverage robot accounts for pushing images from CI/Build pipelines.
+
+
+
+## Decision
+
+### What
+
+* Per workspace, setup a new org “quay.io/unique-org-david/…”
+* Per component, setup a new new repo “quay.io/unique-org-david/unique-component-repo”
+* Use User’s Quay.io API token to manage the org/repo. Short-term, we'll use a pre-configured Quay.io API token associated with AppStudio to create the org/repo till we 
+ figure out how to determinstically map a user in AppStudio to a user in Quay.io.
+* Generate a robot account token scoped to the relevant repository and persist it in the user's workspace for the image build and push process to consume.
+
+
+### How
+
+#### Long-term
+* (One-time) Use an OAuth flow to get David’s Quay.io API token if not previously pulled
+* Create a new Quay.io org owned by David using David’s API token “quay.io/unique-org-david”
+* Create a new image repo for “foo” using David’s API token “quay.io/unique-org-david/appname-foo”
+* Optionally, if needed, make the image repo public using the David’s API token for easy image pulls.
+* Configure the robot account “appstudio” in “quay.io/unique-org-david” to be able to push to “quay.io/unique-org-david/appname-foo” Invalidate/Delete/Prune David’s API token. 
+
+
+#### Short-term
+
+##### Setup
+* Configure the a pre-configured Quay.io API token for repo/org management, also known as the OAuth Access token. From a security perspective, this would be treated as a typical service account token
```

Comment:
```suggestion
* Configure a pre-configured Quay.io API token for repo/org management, also known as the OAuth Access token. From a security perspective, this would be treated as a typical service account token
```